### PR TITLE
feat: add base parameter to /rates API

### LINKED ISSUE #1524: Allow `/rates` API to take in a `base` parameter
Currently, there is no way for the ASE to specify the base currency for the `/rates` API. 
In the MASE, we are returning an object like:
```js
    {
      base: 'USD',
      rates: {
        EUR: 1.1602,
        ZAR: 17.3792
      }
    }
```
However, the `base` would need to be different based on what is passed in the `/rates` API, eg:
`http://cloud-nine-wallet/rates?base=EUR` would return
```js
    {
      base: 'EUR',
      rates: {
        USD: 0.8619,
        ZAR: ...
      }
    }
```

make sure to update the code to the MASE to handle this new parameter as well.

### PR DESCRIPTION
**Pull request by mkurapov**
## Changes proposed in this pull request

- Making sure rates service can handle different exchange rates with multiple decimal points: before, there would be a bug where the currency conversion would fail because BigInt could not support decimal place results that would arise from exchange rates with more decimals than the assetScale
- Adding `base` parameter to `/rates`, updating MASE to properly return with the `base` param

## Context

- Fixes #1524 

## Checklist

- [x] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [x] Documentation added
- [x] Make sure that all checks pass

### CODE REVIEW

File: packages/backend/src/rates/util.test.ts [hunk 1]
```diff
@@ -1,60 +1,64 @@
...
-      ).toBe(50n)
-      // Round down
-      expect(
-        convert({
-          exchangeRate: 0.5,
-          sourceAmount: 101n,
-          sourceAsset: asset(9),
-          destinationAsset: asset(9)
-        })
-      ).toBe(50n)
```
Comment (mkurapov) [inline packages/backend/src/rates/util.test.ts:32]: Rounding up now instead, see https://github.com/interledger/rafiki/pull/1527/files#diff-1eb7ceb504e87d9c310dc012970880799a92b90620ae8a3c2f7d40cafedfb2c0R12

File: packages/backend/src/rates/util.ts [hunk 2]
```diff
@@ -12,11 +12,8 @@ export interface Asset {
...
 export function convert(opts: ConvertOptions): bigint {
-  const maxScale = Math.max(opts.sourceAsset.scale, opts.destinationAsset.scale)
-  const shiftUp = 10 ** maxScale
   const scaleDiff = opts.destinationAsset.scale - opts.sourceAsset.scale
   const scaledExchangeRate = opts.exchangeRate * 10 ** scaleDiff
-  return (
-    (opts.sourceAmount * BigInt(scaledExchangeRate * shiftUp)) / BigInt(shiftUp)
-  )
+
+  return BigInt(Math.round(Number(opts.sourceAmount) * scaledExchangeRate))
```
[the remark this answers is not in the document]
Comment (raducristianpopa) [reply packages/backend/src/rates/util.ts:18]: Do we need to be concerned about exceeding the `MAX_SAFE_INTEGER` (in the `Number(opts.sourceAmount)` case)?

**Review by raducristianpopa — APPROVED** [review]

File: packages/backend/src/rates/util.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mkurapov) [reply packages/backend/src/rates/util.ts:18]: I'm quite certain we will never hit payment amounts over `9007199254740991` in any realistic scenario, but I can add a log just to be extra safe

[the remark this answers is not in the document]
Comment (mkurapov) [reply packages/backend/src/rates/util.ts:18]: We don't have any tooling right now for proper alerting/error management, and I feel like a log would just get swallowed. Going to merge this in for now.

**Pull request merged**